### PR TITLE
Target Java 1.20.4

### PR DESCRIPTION
**Pull request by Konicai**
Targets Java Edition 1.20.4 protocol, and drops support for Bedrock protocol below 1.20.40

Bedrock 1.20.40 - 1.20.5x is currently supported.

### CODE REVIEW

File: core/src/main/java/org/geysermc/geyser/registry/populator/ItemRegistryPopulator.java [hunk 1]
```diff
@@ -664,4 +620,29 @@ private static void registerFurnaceMinecart(int nextFreeBedrockId, List<Componen
         builder.putCompound("components", componentBuilder.build());
         componentItemData.add(new ComponentItemData("geysermc:furnace_minecart", builder.build()));
     }
+
+    /**
+     * Backwards-maps the items of 1.19.50 (630) to 1.19.40 (622)
```
Comment (onebeastchris) [inline core/src/main/java/org/geysermc/geyser/registry/populator/ItemRegistryPopulator.java:625]: 1.20.50/1.20.40 :p

File: README.md [hunk 2]
```diff
@@ -14,7 +14,7 @@ The ultimate goal of this project is to allow Minecraft: Bedrock Edition users t
 
 Special thanks to the DragonProxy project for being a trailblazer in protocol translation and for all the team members who have joined us here!
 
-### Currently supporting Minecraft Bedrock 1.20.0 - 1.20.50 and Minecraft Java 1.20.2
+### Currently supporting Minecraft Bedrock 1.20.0 - 1.20.50 and Minecraft Java 1.20.3
```
Comment (onebeastchris) [inline README.md:17]: 1.20.40-1.20.50

File: core/src/main/java/org/geysermc/geyser/network/UpstreamPacketHandler.java [hunk 3]
```diff
@@ -232,6 +232,9 @@ public PacketSignal handle(ResourcePackClientResponsePacket packet) {
                     stackPacket.getExperiments().add(new ExperimentData("data_driven_items", true));
                 }
 
+                // Required for experimental 1.21 features
+                stackPacket.getExperiments().add(new ExperimentData("updateAnnouncedLive2023", true));
```
Comment (onebeastchris) [inline core/src/main/java/org/geysermc/geyser/network/UpstreamPacketHandler.java:236]: Does BDS send the experiment in both experiment lists (stackpacket and startgame) too?

File: core/src/main/java/org/geysermc/geyser/registry/populator/ItemRegistryPopulator.java (hunk 1 above)
Comment (Konicai) [reply core/src/main/java/org/geysermc/geyser/registry/populator/ItemRegistryPopulator.java:625]: Not sure what you mean - this is mapping 1.19.50 block entries to 1.19.40 block entries - the raw bedrock nbt

**Review by Camotoy — APPROVED** [review]

**Review by Kas-tle — APPROVED** [review]

File: core/src/main/java/org/geysermc/geyser/network/GameProtocol.java [hunk 4]
```diff
@@ -63,19 +60,12 @@ public final class GameProtocol {
...
-            .build());
-        SUPPORTED_BEDROCK_CODECS.add(Bedrock_v618.CODEC.toBuilder()
-            .minecraftVersion("1.20.30/1.20.32")
-            .build());
         SUPPORTED_BEDROCK_CODECS.add(Bedrock_v622.CODEC.toBuilder()
             .minecraftVersion("1.20.40/1.20.41")
             .build());
-        SUPPORTED_BEDROCK_CODECS.add(DEFAULT_BEDROCK_CODEC);
+        SUPPORTED_BEDROCK_CODECS.add(DEFAULT_BEDROCK_CODEC.toBuilder()
+            .minecraftVersion("1.20.50/1.20.51")
```
Comment (onebeastchris) [inline core/src/main/java/org/geysermc/geyser/network/GameProtocol.java:67]: Wasn't there something about not having slashes in the latest supported version, or am I misremembering something

File: bootstrap/fabric/build.gradle.kts [hunk 5]
```diff
@@ -119,7 +119,7 @@ modrinth {
     syncBodyFrom.set(rootProject.file("README.md").readText())
 
     uploadFile.set(tasks.getByPath("remapModrinthJar"))
-    gameVersions.addAll("1.20.2")
+    gameVersions.addAll("1.20.4")
```
Comment (onebeastchris) [inline bootstrap/fabric/build.gradle.kts:122]: please also bump the version in fabric.mod.json

**Review by onebeastchris — APPROVED** [review]

File: core/src/main/java/org/geysermc/geyser/network/GameProtocol.java (hunk 4 above)
Comment (Konicai) [reply core/src/main/java/org/geysermc/geyser/network/GameProtocol.java:67]: That was for pinging stuff right? I think that issue might've been just in regards to DEFAULT_BEDROCK_CODEC. Whereas this just a list that is used to create a message of what is supported

Comment (onebeastchris) [reply core/src/main/java/org/geysermc/geyser/network/GameProtocol.java:67]: Ah, excellent. Thanks for double-checking :)

**Pull request merged**